CONTRIBUTING.md

```diff
@@ -0,0 +1,53 @@
+# Contributing to ntcore
+
+So you want to contribute your changes back to ntcore. Great! We have a few contributing rules that will help you make sure your changes will be accepted into the project. Please remember to follow the rules written here, and behave with Gracious Professionalism.
+
+- [General Contribution Rules](#general-contribution-rules)
+- [What to Contribute](#what-to-contribute)
+- [Coding Guidelines](#coding-guidelines)
+- [Submitting Changes](#submitting-changes)
+    - [Pull Request Format](#pull-request-format)
+    - [Merge Process](#merge-process)
+- [Licensing](#licensing)
+
+## General Contribution Rules
+
+- Everything in the library must work for the 3000+ teams that will be using it.
+- We need to be able to maintain submitted changes, even if you are no longer working on the project.
+- Excluding bug fixes, changes in one language generally need to have corresponding changes in other languages.
+    - Substantial changes often need to have corresponding LabVIEW changes. To do this, we will work with NI on these large changes.
+    - Protocol changes must be carefully considered. Please open issues for protocol enhancements so we can discuss first.
+- Changes should have tests.
+- Code should be well documented.
+    - This often involves ScreenSteps. To add content to ScreenSteps, we will work with you to get the appropriate articles written.
+
+## What to Contribute
+
+- Bug reports and fixes
+    - We will generally accept bug fixes without too much question. If they are only implemented for one language, we will implement them for any other necessary languages. Bug reports are also welcome, please submit them to our GitHub issue tracker.
+- While we do welcome improvements to the API, there are a few important rules to consider:
+    - Features must be added to both the C++ API and the Java API, with rare exceptions.
+    - During competition season, we will not merge any new feature additions. We want to ensure that the API is stable during the season to help minimize issues for teams.
+    - Ask about large changes before spending a bunch of time on them! You can create a new issue on our GitHub tracker for feature request/discussion and talk about it with us there.
+    - Features that make it easier for teams with less experience to be more successful are more likely to be accepted.
+    - Features in ntcore should be broadly applicable to all teams. Anything that is team specific should not be submitted.
+
+## Coding Guidelines
```

Comment:
@PeterJohnson, I'm not sure what coding guidelines you used for this project. Point me at them, and I'll update this section.